sdk/core/core-tracing/src/interfaces.ts

```diff
@@ -248,6 +248,16 @@ export interface TracingSpan {
    * Depending on the span implementation, this may return false if the span is not being sampled.
    */
   isRecording(): boolean;
+
+  /**
+   * Adds an event to the Span.
+   *
+   * @param name - the name of the event.
+   * @param attributesOrStartTime - the attributes that will be added; these are
+   *     associated with this event. Can be also a start time
+   * @param startTime - start time of the event.
+   */
+  addEvent(name: string, attributesOrStartTime?: unknown, startTime?: unknown): void;
```

Comment:
https://github.com/Azure/azure-sdk-for-js/pull/31162/files will add support for this which you can then use (once merged) - you will need to use ^1.1.3 once it is published